test/kernel_apps.js

```diff
@@ -314,8 +313,9 @@ contract('Kernel apps', accounts => {
             await kernel.setApp(APP_BASE_NAMESPACE, appId, appCode1.address)
             assert.equal(appCode1.address, await kernel.getApp(appSetId))
 
-            const appProxy = await kernel.newPinnedAppInstance(appId, '0x0')
-            assert.equal(appCode1.address, await kernel.getApp(appSetId))
+            return assertRevert(async () => {
+                const appProxy = await kernel.newPinnedAppInstance(appId, '0x0')
+            })
```

Comment:
For these new revert cases here and above for `.newAppInstance()`, it might be worth changing the name of the test to better reflect the test :).